phoebe/solverbackends/solverbackends.py

```diff
@@ -2247,6 +2261,240 @@ def valid_options(self):
+            if kwargs.get('deriv_method') == 'symmetric':
+                nforward_models += 2*(len(params_uniqueids)-nanalytic_params)
+            else:
+                nforward_models += len(params_uniqueids)-nanalytic_params
+            if kwargs.get('expose_lnprobabilities'):
+                nforward_models += 1
+            _dc_pbar = _tqdm(total=nforward_models)
+            _use_progressbar = True
+        else:
+            _use_progressbar = False
+
+        datasets_enabled = b.filter(qualifier='enabled', value=True, compute=compute, context='compute', **_skip_filter_checks).datasets
+        # limit to LCs/RVs
+        datasets_enabled = b.filter(dataset=datasets_enabled, kind=['lc', 'rv'], context='dataset', **_skip_filter_checks).datasets
+
+        # need to ensure we access the obs and sigma arrays in the same order
+        obs_params = b.filter(qualifier=['fluxes', 'rvs'], dataset=datasets_enabled, context='dataset', **_skip_filter_checks).to_list()
+        obs_data = np.concatenate([p.get_value() for p in obs_params])
+        obs_sigmas = np.concatenate([b.get_parameter(qualifier=['sigmas'], dataset=p.dataset, component=p.component, context='dataset', **_skip_filter_checks).get_value() for p in obs_params])
+
+        if len(obs_sigmas) != len(obs_data):
+            raise ValueError("sigmas must be provided for all enabled lc and rv datasets")
+
+        A = np.empty(shape=(len(obs_data), len(params_uniqueids)))
+        V = np.diag(1./obs_sigmas)
+
+        deriv_method = kwargs.get('deriv_method')
+        b_solver = _bsolver(b, solver, compute, [])
+
+        def _get_residuals_and_interp_model(b_solver, model, obs_params):
+            # for loop isn't ideal here, but we shouldn't be looping over too many datasets
+            # to optimize this, we would need to ensure that calculate_residuals returns
+            # the concatenated lists in a predictable and reliable order
+            xi = np.array([])
+            obs_baseline = np.array([])
+            for p in obs_params:
+                if p.component == '_default':
+                    continue
+                resid, interp_model = b_solver.calculate_residuals(model=model, dataset=p.dataset, component=p.component, return_interp_model=True, as_quantity=False)
+                xi = np.append(xi, resid)
+                obs_baseline = np.append(obs_baseline, interp_model)
+            return xi, obs_baseline
+
+        baseline_model = b_solver.run_compute(model='baseline', overwrite=True)
+        xi, obs_baseline = _get_residuals_and_interp_model(b_solver, 'baseline', obs_params)
+        if _use_progressbar:
+            _dc_pbar.update(1)
+
+        for k, (uniqueid, value, step) in enumerate(zip(params_uniqueids, p0, steps)):
+            param = b_solver.get_parameter(uniqueid=uniqueid, **_skip_filter_checks)
+            # analytical derivatives:
+            if param.qualifier == 'pblum':
+                A[:,k] = 1
+                continue
```

Comment:
is it worth adding another few simple analytical derivatives here or defer that to a later time?

Reply:
I'd vote to defer.